app/javascript/helpers/ajaxRequestHelper.js

```diff
@@ -0,0 +1,73 @@
+import {XMLHttpRequest} from 'xmlhttprequest';
+
+export class AjaxRequest {
+  constructor(url, method, success) {
+    this.url = url;
+    this.method = method;
+
+    this.errorCallback = () => {
+    };
+
+    if (success == null) {
+      this.successCallback = success;
+    } else {
+      this.successCallback = () => {
+      };
+    }
+
+    this._sendRequest();
+  }
+
+  loadResultIn(elementId) {
+    this.loadInElementId = elementId;
+
+    return this;
+  }
+
+  error(errorCallback) {
+    this.errorCallback = errorCallback;
+
+    return this;
+  }
+
+  // private
+
+  _sendRequest() {
+    const xhr = new XMLHttpRequest();
+
+    this._addErrorListeners(xhr);
+
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4) {
+        this._onReadyStateDone(xhr);
+      }
+    };
+
+    xhr.open(this.method, this.url);
+    xhr.send();
+  };
+
+  _onReadyStateDone(xhr) {
+    if (xhr.status >= 200 && xhr.status < 300) {
+      this._onSuccess(xhr);
+    } else {
+      this.errorCallback(xhr);
+    }
+  }
+
+  _onSuccess(xhr) {
+    this.result = xhr.response;
+
+    if (this.loadInElementId != null) {
+      document.getElementById(this.elementId).innerHTML = this.result;
```

Comment:
I am not where we are using the `hotwire/turbo` feature in this PR 🤔  . We are writing plain JS to request and appending the HTML response to make it SPA. I don't know much  `hotwire/turbo` but should not it handle this stuff itself without writing JS ourselves? 🤔

Reply:
@junan you are right, this file has nothing to do here. It seems like the rebase did not bring this file deletion from `chore/layout-sidebar` branch. This is probably because I did a simple rebase instead of a rebase onto? 
I will investigate the logs. Meanwhile, deleting this file. 

FYI: Turbo does not happened in Javascript file, only in the views with 
- `<%= turbo_frame_tag [...] %>`
- some `data-turbo` attributes

Reply:
yes, confirmed, the rebase of nested branch was the issue. I'm currently studying more in depth `rebase onto` and logs in order to provide with proper & clean rebased for next time. Sorry once again for that =/